### PR TITLE
Composite types documentation

### PR DESCRIPTION
**Pull request by keerlu**
Adding the information in [this Notion doc](https://www.notion.so/prismaio/Using-Embedded-Types-d1a008e84d76415dac9fc1d23432083a) to our docs:

- Schema information added to the data model Concepts page
- Client information added in a new section in Concepts > Components > Client

### CODE REVIEW

File: content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+<Admonition type="info">
+
+Embedded documents are available in Preview in versions 3.10.0 and later.
+
+</Admonition>
+
+<Admonition type="warning">
+
+Embedded documents are currently only available on MongoDB.
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx]: Should probably have something about how to enable this. Preview will be short for this feature though (~6wks), so this will be temporary

Similar to: https://deploy-preview-2858--prisma2-docs.netlify.app/docs/concepts/components/prisma-client/full-text-search#enabling-full-text-search

File: content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+- **Embedding a record**, by storing a record within another record.
+
+Prisma has great [support for relations](/concepts/components/prisma-schema/relations), but support for embedded data has previously been limited to the following options:
+
+- Using [scalar lists](/concepts/components/prisma-client/working-with-fields/working-with-scalar-lists-arrays) to store multiple values in a single field. Only lists of scalar fields are supported.
+- Using [`Json` fields](/concepts/components/prisma-client/working-with-fields/working-with-json-fields) to store JSON data within a single field. `Json` fields are untyped.
+
+Embedded documents introduce a new way to embed data:
+
+- **Embedded documents** allow you to define new object types. Embedded documents are structured and typed, similar to how models are structured and typed today. By understanding the contents of the embedded data, we can offer a richer set of operations you can perform on that data.
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx]: nit: Feels a bit weird having a list with one item.

File: content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  city   String
+  zip    String
+}
+```
+
+In this schema, the `Product` model has a `Photo[]` embedded document, and the `Order` model has two embedded `Address` types. The `shippingAddress` is required, but the `billingAddress` is optional.
+
+### Creating embedded documents
+
+The value of an embedded document can be set within a `create` or `createMany` method using the new `set` method. For example, you can use `set` within `create` to model the case where an `order` contains one `shipping address`:
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx]: IMO, what operations you can perform under what circumstances are the meat of this page:

Right now it feels like these are hidden within the paragraph.

I like what we do with the include & select: 
CleanShot 2022-02-15 at 12 15 54

There may be some redundancy with the Client API Reference though, but I guess my mental model is this page explains how you operate on embedded documents, whereas the reference is more of a handbook when you already know.

**Review by matthewmueller — COMMENTED** [review]
Great start!

File: content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx (hunk 3 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx]: (Reminder for me: screenshot is from https://www.prisma.io/docs/concepts/components/prisma-client/select-fields )

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 4]
```diff
@@ -901,6 +901,59 @@ enum Role {
...
+}
+```
+
+In this case, the `Product` model has an embedded `Photo[]` type.
+
+### Considerations when using composite types:
+
+Composite types currently have some limitations:
+
+- There is limited support for [attributes](/reference/api-reference/prisma-schema-reference#attributes) within composite types. Only `@default`, `@map` and native types (e.g. `@db.ObjectId`) are supported. The `@unique`, `@id`, `@relation`, `@ignore`, `@updatedAt` and `@createdAt` attributes are not supported.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: - Is this really a limitation or just a difference to models and how they work?
- Are native types really not supported?
- `@relation` is covered by the list item below.

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 5]
```diff
@@ -899,6 +899,61 @@ enum Role {
...
+
+</Admonition>
+
+<Admonition type="warning">
+
+Composite types are currently only available on MongoDB.
+
+</Admonition>
+
+Composite types (known as [embedded documents](https://docs.mongodb.com/manual/core/data-model-design/#std-label-data-modeling-embedding) in MongoDB) provide support for embedding records inside other records, by allowing you to define new object types. Composite types are structured and typed in a similar way to models.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: I think we never use the term "record" this way.

Also:
```suggestion
Composite types (known as [embedded documents](https://docs.mongodb.com/manual/core/data-model-design/#std-label-data-modeling-embedding) in MongoDB) provide support for embedding records inside other records, by allowing you to define new object types. Composite types are structured and typed in a similar way to [models](#defining-models).
```

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 6]
```diff
@@ -899,6 +899,61 @@ enum Role {
...
+id     String  @id
+name   String
+photos Photo[]
+}
+
+type Photo {
+height Int
+width  Int
+url    String
+}
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]:
```suggestion
model Product {
  id     String  @id
  name   String
  photos Photo[]
}

type Photo {
  height Int
  width  Int
  url    String
}
```

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 7]
```diff
@@ -899,6 +899,61 @@ enum Role {
...
+```
+
+In this case, the `Product` model has an embedded `Photo[]` type.
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: The `[]` is not really part of the type, but defines that it is a list of `Photo` types that can be stored in `photos`. Might be worth mentioning here.

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 8]
```diff
@@ -899,6 +899,61 @@ enum Role {
...
+
+### Considerations when using composite types:
+
+Composite types currently have some limitations:
+
+- There is limited support for [attributes](/reference/api-reference/prisma-schema-reference#attributes) within composite types. Only `@default`, `@map` and native types (e.g. `@db.ObjectId`) are supported. The `@unique`, `@id`, `@relation`, `@ignore`, `@updatedAt` and `@createdAt` attributes are not supported.
+- [Relations](/concepts/components/prisma-schema/relations) cannot exist inside a composite type
```
Comment (janpio) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: This might be a temporary limitation, not sure what our plans there are.

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 9]
```diff
@@ -0,0 +1,530 @@
+---
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: The current position in the navigation is a bit weird . the things after `CRUD` are also normal model methods like `CRUD` itself.

The first "different" one is `Advanced Type Safety`. So maybe put it in front of that?

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 10]
```diff
@@ -0,0 +1,530 @@
...
+    price: 59.99
+    // Scalar lists that we already support
+    colors: ["Red", "Green"],
+    sizes: ["Small", "Medium", "Large"]
+    // New composite type
+    photos: {
+      set: [
+				{ height: 100, width: 200, url: "1.jpg" },
+				{ height: 100, width: 200, url: "2.jpg" }
+      ]
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]:
```suggestion
      set: [
        { height: 100, width: 200, url: "1.jpg" },
        { height: 100, width: 200, url: "2.jpg" }
      ]
```

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 11]
```diff
@@ -0,0 +1,530 @@
...
+      price: 59.99
+      // Scalar lists that we already support
+      colors: ["Red", "Green"],
+      sizes: ["Small", "Medium", "Large"]
+      // New Composite type
+      photos: {
+        set: [
+				  { height: 100, width: 200, url: "1.jpg" },
+				  { height: 100, width: 200, url: "2.jpg" }
+        ]
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]:
```suggestion
        set: [
          { height: 100, width: 200, url: "1.jpg" },
          { height: 100, width: 200, url: "2.jpg" }
        ]
```

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 12]
```diff
@@ -0,0 +1,530 @@
...
+      // Scalar lists that we already support
+      colors: ["Blue", "Red"],
+      sizes: ["Large", "XLarge"]
+      // New composite type
+      photos: {
+        set: [
+				  { height: 100, width: 200, url: "1.jpg" },
+				  { height: 150, width: 200, url: "4.jpg" }
+				  { height: 200, width: 200, url: "5.jpg" }
+        ]
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]:
```suggestion
        set: [
          { height: 100, width: 200, url: "1.jpg" },
          { height: 150, width: 200, url: "4.jpg" }
          { height: 200, width: 200, url: "5.jpg" }
        ]
```

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 13]
```diff
@@ -0,0 +1,530 @@
...
+
+```ts
+const order = await prisma.order.update({
+  where: {
+    id: 10
+  }
+  data: {
+    billingAddress: {
+			// Unset the billing address
+      // Removes "billingAddress" field from order
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]:
```suggestion
    billingAddress: {
      // Unset the billing address
      // Removes "billingAddress" field from order
```

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 14]
```diff
@@ -0,0 +1,530 @@
...
+
+```ts
+const deleteProduct = await prisma.product.deleteMany({
+  where: {
+    size: 'Small',
+  },
+})
+```
+
+## Filtering composite types with <inlinecode>find</inlinecode> and <inlinecode>findMany</inlinecode>
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Can `find` and `findMany` only used with filters?

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 15]
```diff
@@ -0,0 +1,530 @@
...
+}
+
+type Address {
+  street String
+  city   String
+  zip    String
+}
+```
+
+In this schema, the `Product` model has a `Photo[]` composite type, and the `Order` model has two embedded `Address` types. The `shippingAddress` is required, but the `billingAddress` is optional.
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]:
```suggestion
In this schema, the `Product` model has a `Photo[]` composite type, and the `Order` model has two composite `Address` types. The `shippingAddress` is required, but the `billingAddress` is optional.
```

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 16]
```diff
@@ -0,0 +1,530 @@
...
+
+## Considerations when using composite types
+
+There are currently some limitations when using composite types in the Prisma Client:
+
+- [`findUnique`](/reference/api-reference/prisma-client-reference#findunique) can't filter on composite types
+- [`aggregrate`](/concepts/components/prisma-client/aggregation-grouping-summarizing#aggregate), [`groupBy`](/concepts/components/prisma-client/aggregation-grouping-summarizing#group-by), [`count`](/concepts/components/prisma-client/aggregation-grouping-summarizing#count) don’t support embedded operations.
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Seems like since we're going with composite types, we should also go with composite operations.

```suggestion
- [`aggregrate`](/concepts/components/prisma-client/aggregation-grouping-summarizing#aggregate), [`groupBy`](/concepts/components/prisma-client/aggregation-grouping-summarizing#group-by), [`count`](/concepts/components/prisma-client/aggregation-grouping-summarizing#count) don’t support composite operations.
```

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 17]
```diff
@@ -0,0 +1,530 @@
...
+enum Color {
+  Red
+  Green
+  Blue
+}
+
+enum Size {
+	Small
+	Medium
+	Large
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]:
```suggestion
  Small
  Medium
  Large
```

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 18]
```diff
@@ -0,0 +1,530 @@
...
+        ]
+      }
+    }
+  ]
+})
+```
+
+## Changing composite types within <inlinecode>update</inlinecode> and <inlinecode>updateMany</inlinecode>
+
+Composite types can be set, updated or removed within a `update` or `updateMany` method. The following section describes the methods available for updating a single type or multiple types at once, and gives examples of each.
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]:
```suggestion
Composite types can be set, updated or removed within an `update` or `updateMany` method. The following section describes the methods available for updating a single type or multiple types at once, and gives examples of each.
```

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 19]
```diff
@@ -0,0 +1,530 @@
...
+
+### Changing a single composite type
+
+Use the new `set`, `unset` `update` and `upsert` methods to change a single composite type:
+
+- Use `set` to set a composite type, overriding any existing value
+- Use `unset` to unset a composite type. Unlike `set: null`, `unset` removes the field entirely.
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Inconsistency around lists having periods.

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 20]
```diff
@@ -0,0 +1,530 @@
...
+- Use `update` to update a composite type
+- Use `upsert` to update an existing composite type if it exists, and otherwise create the composite type
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Perhaps break it down to the underlying operations

```suggestion
- Use `upsert` to `update` an existing composite type if it exists, and otherwise `set` the composite type
```

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 21]
```diff
@@ -0,0 +1,530 @@
...
+
+For example, use `update` to update a required `shipping address` with an `Address` composite type inside an `Order`:
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]:
```suggestion
For example, use `update` to update a required `shippingAddress` with an `Address` composite type inside an `Order`:
```

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 22]
```diff
@@ -0,0 +1,530 @@
...
+## Deleting composite types with <inlinecode>delete</inlinecode> and <inlinecode>deleteMany</inlinecode>
+
+To remove records which embed a composite type, use the `delete` or `deleteMany` methods. This will also remove the embedded composite type.
+
+For example, use `deleteMany` to delete all products with a `size` of `"Small"`. This will also delete any embedded `photos`.
+
+```ts
+const deleteProduct = await prisma.product.deleteMany({
+  where: {
+    size: 'Small',
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Is this correct? In the schema it's `sizes  Size[]`

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 14 above)
Comment (matthewmueller) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx:382]: Hmm, what do you mean?

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 23]
```diff
@@ -0,0 +1,530 @@
...
+const orders = await prisma.order.findMany({
+  where: {
+    shippingAddress: {
+      street: '555 Candy Cane Lane',
+    },
+  },
+})
+```
+
+Use `equals` to filter for orders which match on all fields in the shipping address:
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Maybe it's just me, but I always get confused about the word filtering. Does filtering mean: excluding or including matching results?

I'd prefer we use the word "select".

Use `equals` to select orders which match...

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 24]
```diff
@@ -0,0 +1,530 @@
...
+For example, you can use `equals` to find products with
+
+```ts
+const product = prisma.product.findMany({
+  where: {
+    equals: {
+      photos: [{ url: '1.jpg' }, { url: '2.jpg' }],
+    },
+  },
+  orderBy: {
```
Comment (matthewmueller) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Just a note that we just introduced a new concept here: ordering by composite type.

I'm not sure how best to introduce this in the docs, but we should probably mention it somewhere.

**Review by matthewmueller — COMMENTED** [review]
This is excellent, thanks so much @keerlu!!

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 5 above)
Comment (matthewmueller) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: You mean: "embedding records inside other records"?

Any reason not to? Confusing?

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 8 above)
Comment (matthewmueller) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: Likely temporary, let's see how much people need them.

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 4 above)
Comment (matthewmueller) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: > Are native types really not supported?

Might be a bit wordy, but they are supported. It says: _Only `@default`, `@map` and native types (e.g. `@db.ObjectId`) are supported_

> Is this really a limitation or just a difference to models and how they work?

Some work differently, some could be supported in the future.

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 9 above)
Comment (matthewmueller) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Agreed on positioning, especiallly since it's just for MongoDB right now, it should be a bit less prominent

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 14 above)
Comment (janpio) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: I just skimmed the headline and it seemed weird that I could not `find` or `findMany` without filters. But seems one is actually filtering the models for the ones which include a specific composite type? And not searching for embedded documents directly (where it then would also make sense to be able to query for _all_ of them, with no filters)

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx [hunk 25]
```diff
@@ -0,0 +1,530 @@
...
+})
+```
+
+## Filtering composite types with <inlinecode>find</inlinecode> and <inlinecode>findMany</inlinecode>
+
+Composite types can be filtered within a `find` or `findMany` method. The following section describes the methods available for filtering down to a single type or multiple types, and gives examples of each.
+
+### Filter for one composite type
+
+Use the new `is`, `equals` and `isNot` methods to change a single composite type:
```
Comment (janpio) [inline content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]:
```suggestion
Use the `is`, `equals` and `isNot` methods to change a single composite type:
```

File: content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx (hunk 1 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx]: Thanks, have added this

File: content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx (hunk 2 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx]: This got removed when I rearranged the information

File: content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx (hunk 3 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/059-embedded-types.mdx]: The rewritten version should be closer to the Select Fields page (though there are no links to API reference docs as they don't exist yet...)

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 5 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: Yes I'm also unclear, I though this was how we are using the term 'records'? What's the difference here? https://www.prisma.io/docs/about/prisma-docs/style-guide#record

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 19 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Fixed

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 7 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: Added this in.

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 9 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Unfortunately there's no space between 051-working-with-fields and 052-advanced-type-safety :( 

For now I've put it at 056, let me know how to rearrange the numbering system if you want it elsewhere

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 4 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: I've made this into a list form instead - is this any clearer?

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 5 above)
Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: I was not aware of that. All good then.

The suggestion for the link would still be useful I feel.

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 9 above)
Comment (janpio) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: You can just rename the other files to make space where this can fit. Changing the numbers only influences the position in the navigataion AFAIK, so safe to do.

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 22 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Oops, yep. Does that look correct now?

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 23 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: I can change this to `select`... do you also have a suggestion for the comment above where you say 'Hmm, what do you mean?' (sorry can't see how to reply to this directly)

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 5 above)
Comment (keerlu) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: Yep, thanks, I've added the link in

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx (hunk 4 above)
Comment (matthewmueller) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx]: Looks good on my end!

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 9 above)
Comment (matthewmueller) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: It looks okay where it is now, but as Jan mentioned, changing the filename shouldn't affect anything.

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 23 above)
Comment (matthewmueller) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Hahh... I think a prior comment got deleted. Also just deleted the "Hmm, what do you mean".

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 22 above)
Comment (matthewmueller) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Yep, looks good now!

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 24 above)
Comment (matthewmueller) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Commented out for now.

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 20 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Done

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 9 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Ah, ok, thanks. I think I'll move this after the release as a separate PR... it involves renumbering a few files so it's a bit messy and I'm also nervous of having to fix merge conflicts with the other PRs being merged today.

File: content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx (hunk 14 above)
Comment (keerlu) [reply content/200-concepts/100-components/02-prisma-client/032-composite-types.mdx]: Making a note of this as something to follow up after the release.

File: content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx [hunk 26]
```diff
@@ -882,6 +882,71 @@ enum Role {
...
+
+Composite types are available in Preview in versions `3.10.0` and later.
+
+</Admonition>
+
+<Admonition type="warning">
+
+Composite types are currently only available on MongoDB.
+
+</Admonition>
```
Comment (ruheni) [inline content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx:897]: Would we like to merge the 2 sections into one? 🤔

Comment (janpio) [reply content/200-concepts/100-components/01-prisma-schema/04-data-model.mdx:897]: It's two different ones - an info and a warning. Different color in rendering I think.
